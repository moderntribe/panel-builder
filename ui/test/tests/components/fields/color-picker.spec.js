/* global expect b:true */

import React from 'react';
import { mount } from 'enzyme';
import sinon from 'sinon';
import ColorPicker from 'components/fields/color-picker';

const props =  {
	allow_clear: true,
	input_active: true,
	label: 'Background Color',
	default: '#4a7ef2',
	data: '#4a7ef2',
	swatches: [
		'#4a7ef2',
		'#3c73ef',
		'#3b4664',
		'#5cb85c',
		'#d15e61',
		'#ffa700',
		'#eee',
	],
	description: 'Select your bg color for the body.',
};

const render = (overrideProps) => {
	const colorPicker = mount(<ColorPicker
		{...props}
		{...overrideProps}
	/>);

	return {
		colorPicker,
		buttons: colorPicker.find('button'),
	};
};

describe('SwatchSelect field', () => {
	it('calls componentDidMount', () => {
		sinon.spy(ColorPicker.prototype, 'componentDidMount');
		const { colorPicker } = render();
		expect(ColorPicker.prototype.componentDidMount.calledOnce).toEqual(true);
	});

	it('renders 2 buttons as allow clear is set to true', () => {
		const { buttons } = render();
		expect(buttons.length).toEqual(2);
	});

	it('has a label', () => {
		const { colorPicker } = render();
		expect(colorPicker.find('label').length).toEqual(1);
	});

	it('has a description paragraph', () => {
		const { colorPicker } = render();
		expect(colorPicker.find('p').length).toEqual(1);
	});
});
