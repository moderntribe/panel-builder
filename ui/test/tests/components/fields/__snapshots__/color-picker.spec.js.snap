exports[`Color Picker field renders correctly 1`] = `
<div
  className="field panel-field panel-conditional-field">
  <label
    className="label panel-field-label">
    Background Color
    <span
      className="icon"
      onBlur={undefined}
      onClick={undefined}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined} />
  </label>
  <div
    className="container"
    data-js="modular-content-color-picker">
    <div
      className="site-builder__color-field-label colorLabel hasColor hasInput">
      <div
        className="selectedColor"
        style={
          Object {
            "backgroundColor": "#4a7ef2",
          }
        }>
        <i
          className="site-builder__color-arrow selectedColorArrow dashicons dashicons-arrow-down" />
        <button
          className="panel-button full bare colorTrigger"
          data-id=""
          data-loading={false}
          disabled={false}
          onClick={[Function]}
          title=""
          type="button">
          <span>
            
          </span>
        </button>
        <button
          className="panel-button full bare colorClear"
          data-id=""
          data-loading={false}
          disabled={false}
          onClick={[Function]}
          title=""
          type="button">
          <i
            className="dashicons dashicons-no-alt icon" />
          <span>
            
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
